src/main/java/org/unbescape/html/HtmlEscapeUtil.java

```diff
@@ -433,9 +437,60 @@ static int translateIllFormedCodepoint(final int codepoint) {
             case 0x9A: return 0x0161;
             case 0x9B: return 0x203A;
             case 0x9C: return 0x0153;
+            case 0x9D: return 0xFFFD;
             case 0x9E: return 0x017E;
             case 0x9F: return 0x0178;
-            default: return codepoint;
+
+            case 0x000B: return 0xFFFD;
+            case 0xFFFE: return 0xFFFD;
+            case 0xFFFF: return 0xFFFD;
+            case 0x1FFFE: return 0xFFFD;
+            case 0x1FFFF: return 0xFFFD;
+            case 0x2FFFE: return 0xFFFD;
+            case 0x2FFFF: return 0xFFFD;
+            case 0x3FFFE: return 0xFFFD;
+            case 0x3FFFF: return 0xFFFD;
+            case 0x4FFFE: return 0xFFFD;
+            case 0x4FFFF: return 0xFFFD;
+            case 0x5FFFE: return 0xFFFD;
+            case 0x5FFFF: return 0xFFFD;
+            case 0x6FFFE: return 0xFFFD;
+            case 0x6FFFF: return 0xFFFD;
+            case 0x7FFFE: return 0xFFFD;
+            case 0x7FFFF: return 0xFFFD;
+            case 0x8FFFE: return 0xFFFD;
+            case 0x8FFFF: return 0xFFFD;
+            case 0x9FFFE: return 0xFFFD;
+            case 0x9FFFF: return 0xFFFD;
+            case 0xAFFFE: return 0xFFFD;
+            case 0xAFFFF: return 0xFFFD;
+            case 0xBFFFE: return 0xFFFD;
+            case 0xBFFFF: return 0xFFFD;
+            case 0xCFFFE: return 0xFFFD;
+            case 0xCFFFF: return 0xFFFD;
+            case 0xDFFFE: return 0xFFFD;
+            case 0xDFFFF: return 0xFFFD;
+            case 0xEFFFE: return 0xFFFD;
+            case 0xEFFFF: return 0xFFFD;
+            case 0xFFFFE: return 0xFFFD;
+            case 0xFFFFF: return 0xFFFD;
+            case 0x10FFFE: return 0xFFFD;
+            case 0x10FFFF: return 0xFFFD;
+
+            default: break;
+        }
+        if (codepoint >= 0x0001 && codepoint <= 0x0008) {
```

Comment:
I could also inline this initial `if` branch since it is only a few values. Some of these are large ranges, so I wasn't sure how good of an idea it was to inline all of them as `case`s.